include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -0,0 +1,470 @@
+ *        logical conjunction, disjunction and negation operator.
+ * \ingroup io
+ * \tparam condition_t The parse condition type to be extended with the logical operators.
+ *                     Must satisfy the seqan3::detail::parse_condition_concept.
+ */
+template <typename condition_t>
+class parse_condition
+{
+private:
+
+    //!\brief Auxiliary variable to test if the derived type fulfills the parse_condition_concept.
+    template <typename t>
+    static constexpr bool is_parse_condition = parse_condition_concept<t>;
+
+    //!\brief Friend declaration for other seqan3::detail::parse_condition types.
+    template <typename other_condition_type>
+    friend class parse_condition;
+
+public:
+    /*!\name Constructor, destructor and assignment
+     * \brief Standard constructors are defaulted.
+     * \{
+     */
+    constexpr parse_condition() = default;
+    constexpr parse_condition(parse_condition const &) = default;
+    constexpr parse_condition(parse_condition &&) = default;
+    constexpr parse_condition & operator=(parse_condition const &) = default;
+    constexpr parse_condition & operator=(parse_condition &&) = default;
+    ~parse_condition() =  default;
+    //!\}
+
+    /*!\name Logical operators
+     * \brief Adds logical operators to allow logical disjunction, conjunction and negation on parse conditions.
+     * \{
+     */
+    //!\brief Combines the result of two seqan3::detail::parse_condition via logical disjunction.
+    template <parse_condition_concept rhs_condition_t>
+    constexpr auto operator||(parse_condition<rhs_condition_t> const &) const
+    {
+        // using new_condition_t = decltype(or_fn{cond, rhs.cond});
+        return or_fn<condition_t, rhs_condition_t>{};
+    }
+
+    //!\brief Combines the result of two seqan3::detail::parse_condition via logical conjunction.
+    template <parse_condition_concept rhs_condition_t>
+    constexpr auto operator&&(parse_condition<rhs_condition_t> const &) const
+    {
+        // using new_condition_t = decltype(and_fn<condition_t, rhs_condition_t>);
+        return and_fn<condition_t, rhs_condition_t>{};
+    }
+
+    //!\brief Negates the result of a seqan3::detail::parse_condition
+    constexpr auto operator!() const
+    {
+        // using new_condition_t = decltype(not_fn{cond});
+        return not_fn<condition_t>{};
+    }
+    //!\}
+
+    /*!\name Function call operator
+     * \{
+     */
+    //!\brief Invokes the condition on `val`.
+    template <alphabet_concept value_t>
+    bool operator()(value_t && val) const
+    {
+        return std::invoke(condition_t(), std::forward<value_t>(val));
+    }
+    //!\}
+
+    /*!\name Output functions
+     * \{
+     */
+    //!\brief Returns the message representing this condition as std::string.
+    //!\cond
+    template <typename t = condition_t>
+        requires is_parse_condition<t>
+    //!\endcond
+    std::string message() const
+    {
```

Comment:
By moving the requires after the signature you can avoid the weird "template" header.